nixos/modules/services/web-apps/cryptpad.nix

```
@@ -0,0 +1,213 @@
+        serviceConfig = {
+          User = "cryptpad";
+          Environment = [
+            "CRYPTPAD_CONFIG=${cryptpadConfigFile}"
+            "HOME=%S/cryptpad"
+          ];
+          ExecStart = lib.getExe cfg.package;
+          PrivateTmp = true;
+          Restart = "always";
+          StateDirectory = "cryptpad";
+          WorkingDirectory = "%S/cryptpad";
+        };
+      };
+    }
+    (mkIf cfg.confinement {
+      systemd.services.cryptpad = {
+        serviceConfig.BindReadOnlyPaths = [
+          cryptpadConfigFile
+          # apparently needs proc for workers management
+          "/proc"
+          "/dev/urandom"
+        ] ++ (if ! cfg.settings.blockDailyCheck then [
+          "/etc/resolv.conf"
+          "/etc/hosts"
+        ] else []);
+        confinement = {
+          enable = true;
+          binSh = null;
+          mode = "chroot-only";
+        };
+      };
+    })
+    (mkIf cfg.configureNginx {
+      assertions = [
+        { assertion = cfg.settings.httpUnsafeOrigin != "";
+          message = "services.cryptpad.settings.httpUnsafeOrigin is required";
+        }
+        { assertion = strings.hasPrefix "https://" cfg.settings.httpUnsafeOrigin;
+          message = "services.cryptpad.settings.httpUnsafeOrigin must start with https://";
+        }
+        { assertion = isNull cfg.settings.httpSafeOrigin || strings.hasPrefix "https://" cfg.settings.httpSafeOrigin;
+          message = "services.cryptpad.settings.httpSafeOrigin must start with https:// (or be unset)";
+        }
+      ];
+      services.nginx = {
+        enable = true;
+        recommendedTlsSettings = true;
+
+        # FIXME: Check / compare this with [Nginx module configuration in nixpkgs](https://github.com/NixOS/nixpkgs/blob/nixos-23.11/nixos/modules/services/web-servers/nginx/default.nix).
+        # Find out why Cryptpad has this in their documetation. Does this decrease the security of a Cryptpad install
+        # if not used?
+        # diffie-hellman parameters are used to negotiate keys for your session
+        # generate strong parameters using the following command
+        # ssl_dhparam /etc/nginx/dhparam.pem; # openssl dhparam -out /etc/nginx/dhparam.pem 4096
+        #
+        # sslDhparam = null;
+
+        # FIXME: Check / compare this with [Nginx module configuration in nixpkgs](https://github.com/NixOS/nixpkgs/blob/nixos-23.11/nixos/modules/services/web-servers/nginx/default.nix).
+        # Find out why Cryptpad has this in their documentation. Does this decrease the security of a Cryptpad install
+        # if not used?
+        # replace with the IP address of your resolver
+        # resolver 8.8.8.8 8.8.4.4 1.1.1.1 1.0.0.1 9.9.9.9 149.112.112.112 208.67.222.222 208.67.220.220;
+        #
+        # resolver = {};
+
+        virtualHosts = mkMerge [
+          {
+            "${mainDomain}" = {
+              serverAliases = if isNull cfg.settings.httpSafeOrigin then [ ] else [ sandboxDomain ];
```

```suggestion
              serverAliases = lib.optionals (cfg.settings.httpSafeOrigin != null) [ sandboxDomain ];
```
isNull function deprecated and lib.optionals preferred over if true then [ ... ] else [] see https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#syntax

---

Fixed